### PR TITLE
feat: annotation_match for fluentd metadata plugin configurable in values.yaml

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
###### Description

annotation_match for fluentd metadata plugin configurable in values.yaml

---

###### Testing performed

Example tested configuration:

values.yaml:
```yaml
sumologic:
  collectorName: "kubernetes-test" 
  accessId: ""
  accessKey: ""
  endpoint: ""

fluentd:
  logs:
    containers:
      perContainerAnnotationsEnabled: true
      perContainerAnnotationPrefixes:
        - "sumologic.com/"
        - "tailing-sidecar.sumologic.com/"
        
  metadata:
   annotation_match:
    - "\".*sumologic\\.com.*\""
```

pod1.yaml:
```
apiVersion: v1
kind: Pod
metadata:
  name: pod1
  namespace: sumologic
  annotations:
    sumologic.com/count1.sourceCategory: test-source-category5
    sumologic.com/count2.sourceCategory: test-source-category6
  labels:
    app: pod-with-annotations
spec:
  containers:
  - name: count1
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  - name: count2
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  volumes:
  - name: varlog
    emptyDir: {}
  - name: varlogconfig
    emptyDir: {}
```

pod2.yaml:
```
apiVersion: v1
kind: Pod
metadata:
  name: pod2
  namespace: sumologic
  annotations:
    tailing-sidecar.sumologic.com/count1.sourceCategory: test-source-category3
    tailing-sidecar.sumologic.com/count2.sourceCategory: test-source-category4
  labels:
    app: pod-with-annotations
spec:
  containers:
  - name: count1
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  - name: count2
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  volumes:
  - name: varlog
    emptyDir: {}
  - name: varlogconfig
    emptyDir: {}
```

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -316,6 +316,9 @@ fluentd:
     output: false
 
   metadata:
+    ## Option to control capturing of annotations by metadata filter plugin.
+    annotation_match:
+      - "\"sumologic\\.com.*\""
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:321]: Hmm do we need to escape `\` ?

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:321]: `.` must be escaped in regular expression, `\` must be escaped in yaml

e.g. when I changed expression to this form ` "\".*sumologic\.com.*\""` there is this `error: error converting YAML to JSON: yaml: line 17: found unknown escape character`

**Review by pmalek-sumo — APPROVED** [review]
🚢

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:321]: could we use `'"sumologic.com*"'`?

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:321]: `'"sumologic.com\\.*"'` -> wrong dot is escaped, I can test with this form `'"sumologic\\.com.*"'`

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml]: @sumo-drosiek I've changed to this form: `'"sumologic\\.com.*"'`

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**